Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -3279,7 +3282,9 @@ RefPtr<CSSValue> ComputedStyleExtractor::valueForPropertyInStyle(const RenderSty
             return CSSPrimitiveValue::create(CSSValueAuto);
         return CSSPrimitiveValue::createCustomIdent(style.specifiedLocale());
     case CSSPropertyMarginTop: {
-        if (auto* box = dynamicDowncast<RenderBox>(renderer); box && box->isFlexItem() && box->hasTrimmedMargin(PhysicalDirection::Top))
+        auto* box = dynamicDowncast<RenderBox>(renderer);
+        if (box && rendererCanHaveMarginTrimRareDataSetWithinFormattingContext(*box, MarginTrimType::BlockStart) 
+            && box->hasTrimmedMargin(PhysicalDirection::Top))
```

Comment:
any particular reason for changing box's scope here? Are you planning on using it outside?

Reply:
We were having a conversation about readability with this particular piece of code. I'm ok with adding it back though